libxrdp/xrdp_orders_rail.c

```diff
@@ -536,6 +548,12 @@ xrdp_orders_send_notify_new_update(struct xrdp_orders *self,
     {
         /* ToolTip (variable) UNICODE_STRING */
         num_chars = g_mbstowcs(0, notify_state->tool_tip, 0);
+        num_chars = MIN(num_chars, sizeof(twchar) * (g_strlen(notify_state->tool_tip) + 1));
```

Comment:
Same thing here and below. If that code is correct, let's have a function for it and describe the purpose of that calculation.